### PR TITLE
feat: bundle command

### PR DESCRIPTION
**Pull request by P0lip**
This is a breaking change.
Commands:
- build - old good v5 tsc-based approach
- bundle -> post v5 rollup-based approach

https://github.com/stoplightio/platform-internal/issues/925#issuecomment-582780822 drove me to create this PR.

I tried both build and bundling on tree-list and it worked fine.
https://github.com/stoplightio/tree-list/tree/fix/scripts - it also fixes #925

### CODE REVIEW

**Comment by P0lip** [comment]
> error @semantic-release/commit-analyzer@8.0.1: The engine "node" is incompatible with this module. Expected version ">=10.18". Got "8.17.0"

Are you fine with making v10 a minimum required version?

**Comment by XVincentX** [comment]
I'd even put 12 as minimal version @P0lip

File: rollup.config.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ module.exports = {
   input: path.resolve(BASE_PATH, 'src/index.ts'),
   plugins: [
     typescript({
-      tsconfig: getConfigFile('tsconfig.json'),
+      tsconfig: getConfigFile('tsconfig.build.json'),
```
Comment (XVincentX) [inline rollup.config.js]: Hmm, does any project have such file?

File: templates/pkgs/base.json [hunk 2]
```diff
@@ -19,6 +19,7 @@
   },
   "scripts": {
     "build": "sl-scripts build",
+    "bundle": "sl-scripts bundle",
```
Comment (XVincentX) [inline templates/pkgs/base.json]: What would be the difference between build and bundle?

File: src/commands/bundle/index.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  public static examples = [`$ sl-scripts bundle`];
+
+  protected get commands() {
+    const parsed = this.parse(BundleCommand);
+
+    return [
+      buildCommand(`rollup --config ${getConfigFilePath('rollup.config.js')}`, {
+        rawArgs: parsed.raw,
+        flags: Object.keys(BundleCommand.flags),
+      }),
```
Comment (XVincentX) [inline src/commands/bundle/index.ts]: So if I understand correctly rollup will transpire TypeScript to JavaScript and then use Tsc just to get the declarations out, right?

File: templates/pkgs/base.json (hunk 2 above)
Comment (P0lip) [reply templates/pkgs/base.json]: build = plain tsc, a plain transpilation (same as v5 does)
bundle = creates a single file

File: src/commands/bundle/index.ts (hunk 3 above)
Comment (P0lip) [reply src/commands/bundle/index.ts]: Yeah, that's right.
The old rollup plugin emitted declarations for us, but the new one doesn't support it yet.
As soon as they add support for emitting declarations, we can remove the tsc command

**Comment by P0lip** [comment]
> So build would just transpile the code, while bundle would instead use rollup to minify it as well — is that correct?

Yes, that's pretty much it.

> @P0lip Can you quickly explain what was wrong with the previous approach that led to the tree-list problem?

The problem was that certain projects didn't want bundling.
For instance graphite or elements projects do not export every member from the entrypoint, so a scoped imports are quite normal, i.e.
```
import { blabla } from `@stoplight/foo/bla`
```
With the bundling approach, you are expected to re-export everything you want to expose from the main entrypoint, otherwise you won't be able to import that stuff.
In the end, the fact we didn't support both approaches caused fragmentation that led inability to fix stuff, since projects using <=v5 could not be updated to newer version of scripts, thus couldn't benefit from any other changes made to scripts.

**Pull request merged**